src/formatter-settings/index.ts

```diff
@@ -1,99 +1,356 @@
+            setUserError(err);
+            sendError(err);
+            return false;
+        }
+        const javaExtVersion: string = javaExt.packageJSON.version;
+        if (compareVersions(javaExtVersion, JavaConstants.MINIMUM_JAVA_EXTENSION_VERSION) < 0) {
+            vscode.window.showErrorMessage(`The extension version of 'redhat.java' is too stale. Please install at least ${JavaConstants.MINIMUM_JAVA_EXTENSION_VERSION} and run this command again.`);
+            const err: Error = new Error(`The extension version of 'redhat.java' (${javaExtVersion}) is too stale.`);
+            setUserError(err);
+            sendError(err);
+            return false;
+        }
+        await javaExt.activate();
+        this.checkedRequirement = true;
+        return true;
+    }
+
+    private async initialize(document: vscode.TextDocument, webviewPanel: vscode.WebviewPanel): Promise<boolean> {
+        if (!await this.checkRequirement()) {
+            return false;
+        }
+        if (!await this.checkProfile()) {
+            return false;
+        }
+        this.exampleKind = ExampleKind.INDENTATION_EXAMPLE;
+        const onDidChangeSetting: vscode.Disposable = vscode.workspace.onDidChangeConfiguration(async (e) => {
+            if (e.affectsConfiguration(`java.${JavaConstants.SETTINGS_URL_KEY}`) || e.affectsConfiguration(`java.${JavaConstants.SETTINGS_PROFILE_KEY}`)) {
+                this.onChangeProfile(webviewPanel);
+            } else if (e.affectsConfiguration(VSCodeSettings.TAB_SIZE) || e.affectsConfiguration(VSCodeSettings.INSERT_SPACES) || e.affectsConfiguration(VSCodeSettings.DETECT_INDENTATION)) {
+                await this.updateVSCodeSettings(webviewPanel);
+                this.format(webviewPanel);
+            }
+        });
+        webviewPanel.onDidDispose(() => {
+            onDidChangeSetting.dispose();
+        });
+        const diagnostics = this.updateProfileSettings(document, webviewPanel);
+        this.diagnosticCollection.set(document.uri, diagnostics);
+        await this.updateVSCodeSettings(webviewPanel);
+        this.format(webviewPanel);
+        return true;
+    }
+
+    private onChangeProfile(webviewPanel: vscode.WebviewPanel): void {
+        if (webviewPanel.visible) {
+            vscode.window.showInformationMessage(`Formatter Profile settings have been changed, do you want to reload this editor?`,
+                "Yes", "No").then(async (result) => {
+                    if (result === "Yes") {
+                        vscode.commands.executeCommand("workbench.action.webview.reloadWebviewAction");
+                    }
+                });
+        }
+    }
+
+    private updateProfileSettings(document: vscode.TextDocument, webviewPanel: vscode.WebviewPanel): vscode.Diagnostic[] {
+        this.profileElements.clear();
+        this.profileSettings.clear();
+        this.lastElement = undefined;
+        this.diagnosticCollection.clear();
```

Comment:
If you only update the diagnostics when the custom editor is opened, after closing it, these diagnostics will be always there.